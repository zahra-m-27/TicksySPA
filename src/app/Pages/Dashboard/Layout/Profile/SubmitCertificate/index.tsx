import API from '../../../../../API';
import moment from 'jalali-moment';
import Assets from '../../../../../Assets';
import styles from './styles.module.scss';
import {useEffect, useState} from 'react';
import {Button, message, Spin} from 'antd';
import UserIdentityDto from '../../../../../API/DTOs/UserIdentityDto';
import {useHistory} from 'react-router-dom';

export default function SubmitCertificate() {
  const history = useHistory();
  const [Loading, setLoading] = useState(false);
  const [Attachment, setAttachment] = useState<File>();
  const [Identity, setIdentity] = useState<UserIdentityDto>();

  const onSelectFile = (file: File) => {
    setAttachment(file);
  };

  useEffect(() => {
    setLoading(true);
    getIdentity();
  }, []);

  const getIdentity = () => {
    API.Users.GetIdentity({
      status: new URLSearchParams(history.location.search).get('status') ?? '1',
    })
      .then((response) => setIdentity(response))
      .finally(() => setLoading(false));
  };

  const onUploadCertificate = () => {
    setLoading(true);
    API.Users.UpdateIdentity({
      identifier_image: Attachment,
    })
      .then(() => {
        getIdentity();
      })
      .catch(() => {
        setLoading(false);
        message.error('مشکلی در ارسال مدرک احراز هویت رخ داده است.');
      });
  };

  const LoadingComponent = (
    <div className={styles.container_report}>
      <div className={styles.top_part}>
        <div className={styles.title}>
          <img
            alt=""
            src={Assets.Images.FingerPrint}
            className={styles.report_title_image}
          />
          احراز هویت
        </div>
      </div>
      <div className={styles.report_body}>
        <Spin size="large" />
      </div>
    </div>
  );

  if (!Identity) {
    return LoadingComponent;
  }

  if (Identity.status === '3') {
    return (
      <div className={styles.container_upload}>
        <div className={styles.top_part}>
          <div className={styles.title}>
            <img
              alt=""
              className={styles.title_image}
              src={Assets.Images.FingerPrint}
            />
            احراز هویت
          </div>
        </div>
        <div className={styles.upload_body}>
          <p className={styles.upload_text}>
            لطفا سند احراز هویت خود را بارگذاری کنید
          </p>

          <label htmlFor="attachment" className={styles.attachment_box}>
            <div className={styles.input_box}>
              <img
                alt=""
                src={Assets.Images.Upload}
                className={styles.upload_icon}
              />
              <p className={styles.file_name}>
                {Attachment?.name ?? 'انتخاب فایل'}
              </p>
            </div>
          </label>
          <input
            type={'file'}
            id={'attachment'}
            className={styles.input_file}
            onChange={(e) => {
              e.target.files && onSelectFile && onSelectFile(e.target.files[0]);
            }}
          />

          <Button
            type={'primary'}
            loading={Loading}
            onClick={onUploadCertificate}
            className={styles.submit_button}>
            ثبت
          </Button>
        </div>
      </div>
    );
  }

  if (Loading) {
    return LoadingComponent;
  }

  return (
    <div className={styles.container_report}>
      <div className={styles.top_part}>
        <div className={styles.title}>
          <img
            alt=""
            src={Assets.Images.FingerPrint}
            className={styles.report_title_image}
          />
          احراز هویت
        </div>
      </div>
      <div className={styles.report_body}>
        {Identity.status === '1' ? (
          <p className={styles.right_side}>
            <span>سند شما ثبت شد</span>
            <span>لطفا منتظر پاسخ ما باشین</span>
            <span>از صبر و شکیبایی شما </span>
            <span>سپاسگزاریم</span>
          </p>
        ) : (
          <p className={styles.right_side}>
            <span>سند شما تایید شد</span>
            <span>اکنون دسترسی ایجاد تاپیک در سامانه</span>
            <span>برای شما وجود دارد</span>
            <span>با سپاس</span>
          </p>
        )}
        <div className={styles.divider} />
        <div className={styles.left_side}>
          <div className={styles.request_data}>
            <div className={styles.request_container}>
              <span className={styles.request}>وضعیت درخواست:</span>
              <div className={styles.request_data_container}>
                {Identity.status === '1' ? 'در حال بررسی' : 'تایید شده'}
              </div>
            </div>
            <div className={styles.request_container}>
              <span className={styles.request}>تاریخ درخواست:</span>
              <div className={styles.request_data_container}>
                {moment
                  .utc(Identity.request_time ?? new Date())
                  .local()
                  .locale('fa')
                  .format('YYYY/MM/D')}
              </div>
            </div>
            {Identity.status === '1' && (
              <div className={styles.request_container}>
                <span className={styles.request}>تاریخ انقضا:</span>
                <div className={styles.request_data_container}>
                  {moment
                    .utc(Identity.request_time ?? new Date())
                    .add(1, 'week')
                    .local()
                    .locale('fa')
                    .format('YYYY/MM/D')}
                </div>
              </div>
            )}
          </div>
        </div>
      </div>
    </div>
  );
}
