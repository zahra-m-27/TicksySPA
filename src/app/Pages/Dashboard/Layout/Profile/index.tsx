import API from "../../../../API";
import moment from "jalali-moment";
import { Button, message } from "antd";
import Assets from "../../../../Assets";
import styles from "./styles.module.scss";
import { useHistory } from "react-router-dom";
import React, { useRef, useState } from "react";
import useUser from "../../../../Hooks/useUser";
import SEInput from "../../../../Components/SEInput";

export default function Profile() {
  const history = useHistory();
  const { user, Login } = useUser();
  const [Avatar, setAvatar] = useState<File>();
  const [Loading, setLoading] = useState(false);
  const [Email, setEmail] = useState(user.email);
  const emailRef = useRef<HTMLInputElement>(null);
  const [AvatarUrl, setAvatarUrl] = useState<any>();
  const lastNameRef = useRef<HTMLInputElement>(null);
  const [LastName, setLastName] = useState(user.last_name);
  const [EmailHasError, setEmailHasError] = useState(false);
  const [CanChange, setCanChange] = useState<boolean>(false);
  const [FirstName, setFirstName] = useState(user.first_name);
  const [LastNameHasError, setLastNameHasError] = useState(false);
  const [FirstNameHasError, setFirstNameHasError] = useState(false);

  const onFirstNameEnter = () => {
    if (lastNameRef && lastNameRef.current) {
      lastNameRef.current.focus();
    }
  };

  const onLastNameEnter = () => {
    if (emailRef && emailRef.current) {
      emailRef.current.focus();
    }
  };

  const onUpdateProfile = () => {
    if (!/^\S+@\S+$/.test(Email)) {
      setEmailHasError(true);
    }
    if (!FirstName) {
      setFirstNameHasError(true);
    }
    if (!LastName) {
      setLastNameHasError(true);
    }
    if (!/^\S+@\S+$/.test(Email) || !LastName || !FirstName) {
      return;
    }
    setEmailHasError(false);
    setLastNameHasError(false);
    setFirstNameHasError(false);

    setLoading(true);
    API.Users.UpdateProfile({
      email: Email,
      code: user.code,
      last_name: LastName,
      first_name: FirstName,
    })
      .then((response) => {
        Login(response);
      })
      .catch(() => {
        message.error("اشکالی در بروزرسانی پروفایل رخ داده است.");
      })
      .finally(() => {
        setLoading(false);
        setCanChange(false);
      });
  };

  return (
    <div className={styles.rectangle}>
      <div className={styles.header}>
        پروفايل
        <Assets.SVGs.ProfileIconSVG />
      </div>
      <div className={styles.content}>
        <div className={styles.left}>
          <img
            src={user.avatar ?? Assets.SVGs.UserAvatar}
            className={styles.profile}
            alt=""
          />
          {CanChange && (
            <div className={styles.upload}>
              <label htmlFor="add">
                <Assets.SVGs.AddPhotoSVG className={styles.add_photo} />
              </label>
              <input
                id="add"
                type="file"
                className={styles.upload_image}
                onChange={(e) => {
                  if (e.target.files) {
                    setAvatar(e.target.files[0]);
                    var fr = new FileReader();
                    fr.onload = function () {
                      setAvatarUrl(fr.result);
                    };
                    fr.readAsDataURL(e.target.files[0]);
                  }
                }}
              />
            </div>
          )}
        </div>
        <div className={styles.right}>
          {CanChange ? (
            <div className={styles.change}>
              <p className={styles.small}>نام</p>
              <SEInput
                content={FirstName}
                className={styles.input}
                onEnter={onFirstNameEnter}
                onChangeText={setFirstName}
                hasError={FirstNameHasError}
                innerContainerClassName={styles.input_inner_container}
              />
              <p className={styles.small}>نام خانوادگي</p>
              <SEInput
                ref={lastNameRef}
                content={LastName}
                className={styles.input}
                onEnter={onLastNameEnter}
                onChangeText={setLastName}
                hasError={LastNameHasError}
                innerContainerClassName={styles.input_inner_container}
              />
              <p className={styles.small}>ايميل</p>
              <SEInput
                ref={emailRef}
                content={Email}
                regex={/^\S+@\S+$/}
                onChangeText={setEmail}
                className={styles.input}
                hasError={EmailHasError}
                onEnter={onUpdateProfile}
                innerContainerClassName={styles.input_inner_container}
              />
            </div>
          ) : (
            <>
              <p className={styles.big}>نام و خانوادگي</p>
              <p className={styles.small}>
                {user.first_name + " " + user.last_name}
              </p>
              <p className={styles.big}>ايميل</p>
              <p className={styles.small}>{user.email}</p>
              <p className={styles.big}>تاريخ ثبت نام</p>
              <p className={styles.small}>
                {moment
                  .utc(user.date_joined)
                  .local()
                  .locale("fa")
                  .format("YYYY/MM/D")}
              </p>
            </>
          )}
        </div>
      </div>
      <div className={styles.actions}>
        {!CanChange && (
          <Button
            type="primary"
            onClick={() => history.push("/dashboard/submit-certificate")}
            className={styles.change_button}
          >
            احراز هویت
          </Button>
        )}
        <Button
          type="primary"
          loading={Loading}
          onClick={() => {
            if (CanChange) {
              onUpdateProfile();
            } else {
              setCanChange(!CanChange);
            }
          }}
          className={CanChange ? styles.change_button : styles.record_button}
        >
          {CanChange ? "ثبت" : "تغییر"}
        </Button>
      </div>
    </div>
  );
}
