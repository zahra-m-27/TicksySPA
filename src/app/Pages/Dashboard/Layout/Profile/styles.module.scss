$header-background-color: #265ecd;

.rectangle {
  display: flex;
  overflow: hidden;
  margin: 5% auto 0;
  flex-direction: column;
  border-radius: 50px 10px 0 0;
  width: clamp(450px, 39vw, 600px);
  box-shadow: 1px 2px 2px 1px #7070709c;
}

.header {
  opacity: 0.9;
  color: #fff;
  width: 100%;
  height: 61px;
  display: flex;
  font-size: 16px;
  padding-right: 4%;
  align-items: center;
  justify-content: flex-end;
  border-radius: 50px 10px 0 0;
  background-color: $header-background-color;

  svg:nth-child(1) {
    width: 25px;
    color: #fff;
    height: 24px;
    margin: 0 0 0 15px;
    border-color: #fff;
  }
}

.content {
  display: flex;
  padding-top: 2rem;
  align-items: center;
}

.right {
  flex: 1;
  margin-right: 8%;
}

.left {
  display: table;
  margin-left: 8%;
  position: relative;
}

.big {
  direction: rtl;
  line-height: 1;
  font-size: 24px;
  color: #454647;
  text-align: start;
  font-style: normal;
  font-weight: normal;
  object-fit: contain;
}

.small {
  opacity: 0.7;
  direction: rtl;
  color: #454647;
  font-size: 15px;
  text-align: start;
  font-style: normal;
  object-fit: contain;
  font-weight: normal;
}

.profile {
  margin-top: 10%;
  border-radius: 50%;
  width: clamp(150px, 12vw, 230px);
  height: clamp(150px, 12vw, 230px);
}

.upload {
  bottom: 0;
  right: -50%;
  display: flex;
  position: absolute;
  align-items: center;
  flex-direction: column;
  width: clamp(60px, 7vw, 100px);
  height: clamp(60px, 7vw, 100px);
  transform: translate(-100%, 25%);
}

.add_photo {
  width: 65px;
  height: 65px;
  object-fit: none;
}

.upload_image {
  display: none;
}

.actions {
  display: flex;
  justify-content: space-evenly;
}

.change_button {
  width: 30%;
  opacity: 0.9;
  height: 32px;
  margin-top: 2rem;
  margin-bottom: 3%;
  align-self: center;
  background-color: $header-background-color;
}

.change {
  width: 75%;
  margin-left: auto;
}

.change > input {
  width: 100%;
  opacity: 0.7;
  height: 30px;
  border: none;
  padding: 15px;
  direction: rtl;
  font-size: 10px;
  border-radius: 5px;
}

.record_button {
  width: 30%;
  opacity: 0.9;
  height: 32px;
  margin-top: 2rem;
  margin-bottom: 3%;
  align-self: center;
  background-color: $header-background-color;
}

.input {
  margin-bottom: 10px;
}

.input_inner_container {
  box-shadow: none;
}
