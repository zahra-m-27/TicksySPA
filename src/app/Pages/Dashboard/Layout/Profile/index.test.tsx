import ReactDOM from 'react-dom';
import React from 'react';
import {render} from '@testing-library/react';
import renderer from 'react-test-renderer';
import {MemoryRouter} from 'react-router-dom';
import Profile from './index';

const sample = (
  <MemoryRouter>
    <Profile />
  </MemoryRouter>
);

test('render in ReactDOM', () => {
  const div = document.createElement('div');
  ReactDOM.render(sample, div);
});

it('render header', () => {
  const {getByText} = render(sample);
  getByText(/پروفايل/i);
});

it('render name field', () => {
  const {getByText} = render(sample);
  getByText(/نام و خانوادگي/i);
});

it('render email field', () => {
  const {getByText} = render(sample);
  getByText(/ايميل/i);
});

it('render personal identifier', () => {
  const {getByText} = render(sample);
  getByText(/شماره دانشجویی/i);
});

it('render register date field', () => {
  const {getByText} = render(sample);
  getByText(/تاريخ ثبت نام/i);
});

it('render verification button', () => {
  const {getByText} = render(sample);
  getByText(/احراز هویت/i);
});

it('render submit', () => {
  const {getByText} = render(sample);
  getByText(/تغییر/i);
});

it('snapshot', () => {
  const snapshot = renderer.create(sample).toJSON();
  expect(snapshot).toMatchSnapshot();
});
