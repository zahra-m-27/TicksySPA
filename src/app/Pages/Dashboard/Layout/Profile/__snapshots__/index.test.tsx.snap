// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<div
  className="rectangle"
>
  <div
    className="header"
  >
    پروفايل
    <svg
      height="28.978"
      viewBox="0 0 34.978 34.978"
      width="28.978"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M17.489 0a17.489 17.489 0 1 0 17.489 17.489A17.509 17.509 0 0 0 17.489 0zm0 2.691A14.79 14.79 0 0 1 28.42 27.453c-1.009-1.855-3.978-3.379-7.147-4.036a1.288 1.288 0 0 1-.841-1.892 7.362 7.362 0 0 0 1.513-2.943 4.322 4.322 0 0 0 1.471-2.691c.137-1.482-.294-1.64-.294-1.64.536-1.75.683-8.335-3.489-7.525-.673-1.345-5.087-2.412-7.1 1.219a7.135 7.135 0 0 0-.42 6.18c0 .137-.284-.142-.42.8a5.788 5.788 0 0 0 .841 2.817 1.39 1.39 0 0 0 .673.547 7.312 7.312 0 0 0 1.465 3.111 1.874 1.874 0 0 1-.967 1.892c-3.279.657-6.259 2.181-7.273 4.036A14.8 14.8 0 0 1 17.489 2.691z"
        fill="#fff"
      />
    </svg>
  </div>
  <div
    className="content"
  >
    <div
      className="left"
    >
      <img
        alt=""
        className="profile"
        src=""
      />
    </div>
    <div
      className="right"
    >
      <p
        className="big"
      >
        نام و خانوادگي
      </p>
      <p
        className="small"
      >
         
      </p>
      <p
        className="big"
      >
        ايميل
      </p>
      <p
        className="small"
      >
        
      </p>
      <p
        className="big"
      >
        شماره دانشجویی / کد پرسنلی
      </p>
      <p
        className="small"
      >
        
      </p>
      <p
        className="big"
      >
        تاريخ ثبت نام
      </p>
      <p
        className="small"
      >
        1400/05/10
      </p>
    </div>
  </div>
  <div
    className="actions"
  >
    <button
      className="ant-btn ant-btn-primary change_button"
      data-testid="submit_certificate"
      onClick={[Function]}
      type="button"
    >
      <span>
        احراز هویت
      </span>
    </button>
    <button
      className="ant-btn ant-btn-primary record_button"
      data-testid="edit_button"
      onClick={[Function]}
      type="button"
    >
      <span>
        تغییر
      </span>
    </button>
  </div>
</div>
`;
